nwbinspector/utils.py

```diff
@@ -0,0 +1,26 @@
+"""Authors: Cody Baker and Ben Dichter."""
+import numpy as np
+from collections import defaultdict
+
+global default_checks
+default_checks = {1: defaultdict(list), 2: defaultdict(list), 3: defaultdict(list)}
+
+
+def add_to_default_checks(severity: int, neurodata_type):
+    """Wrap a check function to add it to the list of default checks for that severity and neurodata type."""
+
+    def decorator(check_function):
+        if severity not in [1, 2, 3]:
+            raise ValueError(
+                f"Indicated severity ({severity}) of custom check ({check_function.__name__}) is not in range of 1-3."
+            )
+        default_checks[severity][neurodata_type].append(check_function)
```

Comment:
Rather than just adding the function to the global dict it may be worth considering to create a base class ``NWBInspectorCheck`` and wrapping the function itself with it. The advantage of this approach is that the metadata about the check (e.g,. severity, neurodata_type etc.) are directly associated with the check itself (instead of being only in the global default_checks dict), which will make introspection of checks easier. Another advantage is that this gives all checks a consistent interface while at the same time preserving the simplicity of the decorator for the user.

```
class NWBInsepctorCheck:
     def __init__(self, severity, neurodata_type, name, check):
          self.severity = severity
          self.neurodata_type = neurodata_type
          self.name = check.__name__
          self.check = check
   
    def __call__(self, **kwargs):
          self.check(**kwargs)    
```

Reply:
I've implemented Ben's suggestion setting these as attributes of the function (which I didn't know could be done in python, but is pretty cool); this has the exact same effect of associating critical internal properties directly with their corresponding check function, without needing to do anything more complicated in going between classes, functions, and the decorator.